test-network-function/generic/suite.go

```diff
@@ -653,9 +653,13 @@ func getContext() *interactive.Context {
 func testOwner(podNamespace, podName string) {
 	ginkgo.When("Testing owners of CNF pod", func() {
 		ginkgo.It("Should contain at least one of kind DaemonSet/ReplicaSet", func() {
+			extraInfo := []string{}
 			context := getContext()
```

Comment:
Can we add something like this in to resolve Charlie's point:
```
written := false
// emit informational message(s) in the case of early exit
defer func() {
    if !written {
        tnf.WriteTestExtraInfo(extraInfo)
    }
}()

context := getContext()
tester := owners.NewOwners(defaultTimeout, podNamespace, podName)
extraInfo = append(extraInfo, fmt.Sprintf("Pod %s/%s before execution", podNamespace, podName))
test, err := tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
extraInfo = append(extraInfo, fmt.Sprintf("Pod %s/%s executed", podNamespace, podName))
tnf.WriteTestExtraInfo(extraInfo)
written = true
...
```

Reply:
The defer is in the context of the test and not in the code I supply.

Reply:
I disagree.  You are streaming a slice of messages in this example, and if an error occurs (such as in line 660), then the stream is never flushed, and all context is lost.

Reply:
I am a bit confused by your question, so let me attempt to clarify.  You edited `test-network-function/generic/suite.go` in this Pull Request to include the call to the `WriteTestExtraInfo(...)` API added `pkg/tnf/test.go`.

I am advocating that the Client usage of `WriteTestExtraInfo(...)` in `suite.go` is done in a `defer func(){}` in order to ensure that any intermediate `extraInfo` is written to the claim, regardless of whether an exception occurs in the `Ginkgo.It` block started on line 655.  The `defer func(){}` will be called when the anonymous function on line 655 exits, regardless of whether there is an intermediate error.  Thus any in-flight `extraInfo` strings will be written to the claim.